docs/architecture-decision-records/0020-aggregation-for-tdr-preingest.md

```diff
@@ -0,0 +1,40 @@
+| -------------------------------------------------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Ingest in transfer consignment batches.                              | - The preingest would be a simple transform from BagIt to DR2 BagIt-like.<br>- Reporting of ingest status would easy for humans to understand. | - Size of ingest batches is user defined and unoptimized for the Preservation System.<br>- TDR BagIt packages are `.tar.gz` compressed, we'd need to build a process to unpack as part of the transformation. As consignments get larger we may struggle to do this using Serverless services.<br>- Large transfers, or transfers from systems other than TDR, will require a different solution. |
+| Ingest single records per workflow execution.                        | - Easy to implement.                                                                                                                           | - Extremely inefficient for Preservation System.<br>- Unable to scale easily.<br>- May reach AWS quota limits for the number concurrent workflows.                                                                                                                                                                                                                                                |
+| Store incoming events and trigger ingest using scheduled/cron tasks. | - DR2 can control ingest batch size.                                                                                                           | - May struggle to scale during high load, messages could be held without processing for extended periods.<br>- Requires new database technology to query and update many rows in a single action.<br>- Anti-pattern to event based architecture as events are not triggering consumers.                                                                                                           |

```

Comment:
1. I think a semicolon instead of a comma after "during high load"
2. missing hyphen in "event based"